### DISCUSSION TITLE
Volatility targeting in portfolio optimization.

### DISCUSSION THREAD

**Discussion by NervousEnergy1979** [cvxpy/cvxpy#2180] [Q&A]
Hi, I have an issue where for some reason my problem is not DCP compliant, even though it seems the indivdual expressions are. I have a set of target portfolio weights `w` that are breaking some constraints (concentration limits, leverage etc.), so I have the following problem setup

```
import cvxpy as cv

# get daily variance from annualised std vol target of 10%
vol_targ = (10 / (np.sqrt(252) * 100)) ** 2

x = cv.Variable(len(w))
covar = returns.cov().values  # variance-covariance matrix of asset returns
vol = cv.quad_form(x, covar)  # risk (volatility)

sub_expr1 = cv.sum(cv.abs(x - w))  # want new weights x to be as close as possible to known target weights w
sub_expr2 = cv.abs(vol - vol_targ)  # want risk to be as close as possible to the 10% standard deviation
expr = cv.Minimize(sub_expr1 + sub_expr2)
constraints = [
    cv.sum(x) == 1.0, # fully invested, leverage = 1
    x >= 0.0,  # no shorting, long only
]
prob = cv.Problem(expr, constraints)
prob.solve()

>>> DCPError: Problem does not follow DCP rules. Specifically:
The objective is not DCP. Its following subexpressions are not:
abs(QuadForm(var1284394, [[....
```

As far as I can see, both `sub_expr1` and `sub_expr2` are convex in theory - of course, when I print `sub_expr2` I get the result `Expression(UNKNOWN, NONNEGATIVE, (1, 1))` instead of the expected `Expression(CONVEX, NONNEGATIVE, (1, 1))`, and I do not know why. Can anyone explain to me why this is, or how I should change `sub_expr2` to be convex? The end goal is just to adhere to a target volatility as closely as possible, which is why I try to minimize the distance between `vol` and `vol_targ`. Thank you

**Comment by phschiele** [comment]
@ktattan There is a technical answer to that question, but the implication goes a bit further.
1. The `sub_expr2` results in a constraint that looks like `convex >= constant`. This is not DCP, compare the [DCP ruleset](https://www.cvxpy.org/tutorial/dcp/index.html#dcp-problems).
The constraint is introduced as `abs(x)` is rewritten by CVXPY as `t`, with `-t <= x <= t`. 

3. The implication of this is that you can constrain risk from above, but not from below. Luckily, this is what you want in most cases. If you can have less risk everything else being equal, why would you choose to have more?
